### ISSUE TITLE
Use Svelte?

### ISSUE THREAD

**Issue by mmalmi** [irislib/iris-messenger#113]
Hey Guys, you just speak from my soul 🙏

I just started a fork of Iris https://github.com/worldpeaceenginelabs/METAVERSE-GUN

![image](https://user-images.githubusercontent.com/67427045/173733545-df58dcef-41fa-430f-978f-c3b893226a9d.png)

Iris Messenger is just such a good base to start something like this. Look at all the green checkboxes in readme.md (this is what the iris-community has already accomplished) 🙏 just wow! 🤯

I would love to have the codebase of Iris in Svelte!

Since Svelte isnt a language but a compiler, which compiles into Vanilla JS, its like writing in react or vue for instance, but saving the framework typical boilerplate. (You dont need to learn anything new) Like i transfered js code from react projects to Svelte with just cutting off the uneccasary bs surrounding the actual code 😅 (and i code active for just about 4-6 weeks)

Its source code is easy to understand for developers. Svelte.js gathers the logic (JavaScript), the structure (HTML), and the style (CSS) in the same file. This makes the code easy to read and maintain. Guillermo Rauch, Vercel founder & CEO, has even said Svelte, in his opinion, is a lot easier to learn than React - an environment he’s known perfectly for ages.

Svelte does even safe you the DOM/router stuff. So its basically connecting components via imports, exports and modals only.
Less code, better readability, less bugs, less space, faster execution!

This video hitted me. So simple. And since it is clean, typed js, its easier to reuse code or get a snippet from stackexchange.
https://youtu.be/J5x3OMXjgMc Gun Svelte Chat App

Iris definitly needs a makeover to get more code contributions.

But if not Svelte, then minimum Typescript and markup seperated.
Oh wait, such a coincidence, the last part was just Svelte i described 😅

_Originally posted by @worldpeaceenginelabs in https://github.com/irislib/iris-messenger/issues/108#issuecomment-1155956494_

**Comment by trosel** [comment]
I'd like to hear more about this

> Iris definitly needs a makeover to get more code contributions.

A makeover in terms of UI / design? If so, I could see that. But it just hasn't been a priority yet

A makeover in terms of framework / technology? There are way more React developers and designers in the world than Svelte ones. Why would changing to Svelte attract more?

**Comment by worldpeaceenginelabs** [comment]
I am sooo happy that you take Svelte into consideration 🥳🙏

**Framework as a Compiler: The Future of Web Development?**
https://naturaily.com/blog/why-svelte-is-next-big-thing-javascript-development

I made you guys some example screenshots of my CesiumJS project (not Metaverse-Gun) in VS Code, to let you see, how simple and nicely ordered everything gets with Svelte. 😍

Every file is basically a page/view/component at the same time. (no router/vDOM necessary)
You basically navigate with imports, exports and modals.

The code of each file is seperated into logic (JavaScript), the structure (HTML), and the style (CSS) in the same file. 

```
//file Example.svelte

<script>here goes your logic</script>

<div>here goes your HTML</div>

<style>here goes your style</style>
```

CesiumJS is a good example, because on first glance, even professionals say "ugh, looks complicated".
But its not! At least not with Svelte (again, i am a html, css beginner, coding 4-6 weeks in js)
Imagine what long experienced coders like you could achieve, without all the bloated boilerplate you safe 😉
<br>

Notice that you are in control of the css style for every Svelte file seperatly (page/view/component)

![image](https://user-images.githubusercontent.com/67427045/173848964-527c4986-72d4-416f-8b43-58be60ba2481.png)
This is the App.svelte file, your start view, or most upper layer of abstraction.
You can basically just drop everything here with `<Component/>`
<br>

![image](https://user-images.githubusercontent.com/67427045/173850227-f66cc6c3-6897-4544-a0a9-ae17f3dc2485.png)
See? Everything that belongs too the Cesium globe, i just dropped in the file Cesium.svelte
<br>

![image](https://user-images.githubusercontent.com/67427045/173850037-5d61dbc0-c1b4-4a37-8ace-630a5dfa8e0f.png)
This is the topbar (a picture and some text) (Sdgtopbar.svelte)
<br>

![image](https://user-images.githubusercontent.com/67427045/173850454-fe18b824-af29-4c92-a1b1-53264f5ed3ce.png)
This is the "Coming Soon" text. (Comingsoontext.svelte)
<br>

![image](https://user-images.githubusercontent.com/67427045/173853650-c461f060-4f5c-48eb-bd5b-d8362b77fc0b.png)
This is the modal logic btw (Modalmain.svelte)

Have a liveview of it: https://cloudatlas.club

A word on the bundler:

I prefer Vite, since it is basically a lightspeed fast rollup. 500ms startup time and hot reload. `npm run dev`
Static output for CDN with `npm run build`
WITHOUT any further configuration, right from the start!

So when i start a new Svelte project, i start like `npm create vite@latest`, follow the prompt and choose "svelte", then "svelte-ts" and thats it. https://vitejs.dev/guide/#scaffolding-your-first-vite-project

![image](https://user-images.githubusercontent.com/67427045/173859095-e18d5f0c-0f84-4c39-8faa-9a6fa714b0a6.png)


So fresh and so clean, you can just start to code in Typescript whats in your head from here 😍

**Comment by worldpeaceenginelabs** [comment]
> A makeover in terms of framework / technology? There are way more React developers and designers in the world than Svelte ones. Why would changing to Svelte attract more?

yes. we can optimize the UI anyway, but has you said, it has no priority.

There are way more React developers and designers in the world than Svelte ones, aggree.
But as i said: You can switch instant from React to Svelte, just saving the boilerplate.
Anything else is like you used to. (except some Svelte specifics of course, bindings, exports, but that makes your day easier, not worse) 😉👍

**Comment by worldpeaceenginelabs** [comment]
I love the Iris UI btw, maybe some cosmetics here and there, but nothing drastic.
You guys did an awesome work already!!! 🙏

**Comment by worldpeaceenginelabs** [comment]
I made you another example, closer to your decentralization stack 😁👍

This is about making Gun a back-end for your websites user inputs, like a subscription form for instance.

All nicely in just one file!
(not working yet properly, but due to my own inexperience with Gun only😅)

![image](https://user-images.githubusercontent.com/67427045/173880520-87571f60-f9cb-46ae-94a8-70c8573e70d8.png)

**Comment by trosel** [comment]
@worldpeaceenginelabs there is already a guide for using svelte with gun on the original gun github wiki. 

but i gun also works nicely as the backend data layer for react as well. there is a guide for that too

**Comment by worldpeaceenginelabs** [comment]
> @worldpeaceenginelabs there is already a guide for using svelte with gun on the original gun github wiki.
> 
> but i gun also works nicely as the backend data layer for react as well. there is a guide for that too

thx.🙏 yes i saw them, and I am about to figuring it out with their help (i actually guess i just miss the right var scope here 😅)

let me clarify that i'm not against react 😇
but even as a beginner, i was just able to use the JS code from react apps, with the same result at the end, but minus the boilerplate.

Thats my point here. I am convinced every dev should switch to Svelte, since it is Typescript without the framework agnostic, useless code surrounding react.

React and other frameworks were a great helper on the way to today, but there is something new on the horizon.

And the only thing you need to know and understand is JS/Typescript.

You know JS/Typescript, arent you? ✌️😜

**Comment by worldpeaceenginelabs** [comment]
Ive got it working! Had two times gun.get invoked before. 🤦‍♂️

I am starting to offer 30-60min courses on Fiverr, as of today. For Svelte js beginners and js professionals who want to transfer to Svelte, or want to come back to good "old" Typescript. A decision which gets even honorized with Vanilla JS code at the end🤩
But i am not teaching a language to others, after just 6 weeks of coding actively, of course.

But everybody will think its a language course, but its not.
Its a compiler course, because thats what Svelte is.

And i understand the compiler very well, just not JS yet 😅

So I am not asking you, to change your language. I basically just ask you to change your compiler for iris-messenger actually 😉 (i had to grasp that myself again, just a few minutes ago)

Since the other options you are thinking of seems to be Typescript anyway, you can just drop your Typescript in the red box, and your html file into the green box.

I just drop you the screenshot of my typical desktop as a visual helper and the repo, because you can just test in Typescript whatever you want, to get a first touch with Svelte instant!

Modify the server, test a new script, test the build and load the static files to a cdn maybe?...

Its a fresh `npm create vite@latest`, nothing touched but App.svelte and `npm install gun`
https://github.com/worldpeaceenginelabs/decentralize-subscription-service

I would love to read your feedbacks on how you like to code in Svelte...

![image](https://user-images.githubusercontent.com/67427045/174095893-a7695272-31b6-4106-99d9-2af08c5d3f56.png)

**Comment by sascha1337** [comment]
YES. just YES. i am investigating svelte since longer now, daily, i prefer it 50x more than (p)react

**Comment by worldpeaceenginelabs** [comment]
> YES. just YES. i am investigating svelte since longer now, daily, i prefer it 50x more than (p)react

btw: i prefer Svelte in general, but i found a way to use svelte and preact at the same time.

Svelte (and any other framework) can be compiled to webcomponents, which can be used in Preact (and any other framework) as a component.

This way a transition of Iris from Preact to Svelte could go easy, by adding new Iris features in Svelte, and update the whole Iris code to Svelte step by step with time...

Here a video from Fireship.io which shows how to cross use webcomponents:https://youtu.be/SJeBRW1QQMA

Here another one, more detailed: https://youtu.be/RbTM785zhG4

Two things to take a look on 
https://docs.astro.build/en/getting-started/
and
https://github.com/fireship-io/flamethrower (which is in meme status for now)

And

**Comment by petrikaj** [comment]
Iris is a nostr client now.
